Lib/asyncio/tasks.py

```diff
@@ -104,7 +104,8 @@ def __init__(self, coro, *, loop=None, name=None, context=None):
             raise TypeError(f"a coroutine was expected, got {coro!r}")
 
         if name is None:
-            self._name = f'Task-{_task_name_counter()}'
+            # optimization: defer task name formatting to first get_name
+            self._name = None
```

Comment:
Why do we need to optimize the .py version (which is normally not run since there is a C accelerator version)?

Reply:
we definitely don't need to optimize the python version. I assumed we wanted to maintain parity between the versions, but maybe it doesn't matter to this extent (or my assumption was wrong and parity is not a goal).

I will revert the changes to the python version.